Services/OperationService/Commands/RunBotCommand.cs

```diff
@@ -1,46 +1,79 @@
 
-        public RunBotCommand([FromServices] IRequestClient<RunBotRequest> client, [FromServices] ILogger<RunBotCommand> logger)
+        public RunBotCommand(
+            [FromServices] IRequestClient<RunBotRequest> runClient,
+            [FromServices] IRequestClient<InternalGetBotRulesRequest> getRulesClient,
+            [FromServices] ILogger<RunBotCommand> logger,
+            [FromServices] ICommand<TradeRequest, bool> tradeCommand,
+            [FromServices] ICommand<GetCandlesRequest, IEnumerable<Candle>> candlesCommand,
+            [FromServices] ICommand<GetUserBalanceRequest, UserBalance> balanceCommand,
+            [FromServices] ICommand<GetInstrumentsRequest, IEnumerable<Instrument>> instrumentsCommand)
         {
-            this.client = client;
+
+            this.runClient = runClient;
+            this.getRulesClient = getRulesClient;
             this.logger = logger;
+            this.tradeCommand = tradeCommand;
+            this.candlesCommand = candlesCommand;
+            this.balanceCommand = balanceCommand;
+            this.instrumentsCommand = instrumentsCommand;
         }
 
-        private async Task<bool> RunBot(RunBotRequest request)
+        private async Task RunBot(InternalGetBotRulesRequest getRulesRequest, RunBotRequest runRequest)
         {
             logger.LogInformation("Response from Database Service RunBot method received");
 
-            var response = await client.GetResponse<OperationResult<bool>>(request);
+            var response = await getRulesClient.GetResponse<OperationResult<List<BotRuleData>>>(getRulesRequest);
+
+            var rulesData = OperationResultHandler.HandleResponse(response.Message);
+
+            BotRunner.Run(runRequest, rulesData, tradeCommand, candlesCommand, balanceCommand, instrumentsCommand);
 
-            return OperationResultHandler.HandleResponse(response.Message);
+            await runClient.GetResponse<OperationResult<bool>>(runRequest);
         }
 
-        public async Task<bool> Execute(RunBotRequest request)
+        public async Task<bool> Execute(RunBotRequest runRequest)
         {
+            var getRulesRequest = new InternalGetBotRulesRequest() { BotId = runRequest.BotId };
             try
             {
-                return await RunBot(request);
+                await RunBot(getRulesRequest, runRequest);
+                return true;
             }
-            catch (Exception)
+            catch (NotFoundException)
             {
                 var e = new NotFoundException("Not found bot to run");
-                logger.LogWarning(e, $"{e.Message}, botId: {request.ID}");
+                logger.LogWarning(e, $"{e.Message}, botId: {runRequest.BotId}");
+                throw e;
+            }
+            catch (BadRequestException e)
+            {
+                logger.LogWarning(e, $"{e.Message}, botId: {runRequest.BotId}");
                 throw e;
```

Comment:
maybe we should catch exceptions of all types

Reply:
> maybe we should catch exceptions of all types

Yeah, there's some reason in adding default handler mb